src/interpreter/ByteCode.h

```diff
@@ -30,23 +28,234 @@
+    F(I64TruncF32S, doConvert, float, int64_t, int64_t, float)          \
+    F(I64TruncF32U, doConvert, float, uint64_t, uint64_t, float)        \
+    F(I64TruncF64S, doConvert, double, int64_t, int64_t, double)        \
+    F(I64TruncF64U, doConvert, double, uint64_t, uint64_t, double)      \
+    F(F32ConvertI32S, doConvert, int32_t, float, float, int32_t)        \
+    F(F32ConvertI32U, doConvert, uint32_t, float, float, uint32_t)      \
+    F(F32ConvertI64S, doConvert, int64_t, float, float, int64_t)        \
+    F(F32ConvertI64U, doConvert, uint64_t, float, float, uint64_t)      \
+    F(F64ConvertI32S, doConvert, int32_t, double, double, int32_t)      \
+    F(F64ConvertI32U, doConvert, uint32_t, double, double, uint32_t)    \
+    F(F64ConvertI64S, doConvert, int64_t, double, double, int64_t)      \
+    F(F64ConvertI64U, doConvert, uint64_t, double, double, uint64_t)    \
+    F(I32TruncSatF32S, intTruncSat, float, int32_t, int32_t, float)     \
+    F(I32TruncSatF32U, intTruncSat, float, uint32_t, uint32_t, float)   \
+    F(I32TruncSatF64S, intTruncSat, double, int32_t, int32_t, double)   \
+    F(I32TruncSatF64U, intTruncSat, double, uint32_t, uint32_t, double) \
+    F(I64TruncSatF32S, intTruncSat, float, int64_t, int64_t, float)     \
+    F(I64TruncSatF32U, intTruncSat, float, uint64_t, uint64_t, float)   \
+    F(I64TruncSatF64S, intTruncSat, double, int64_t, int64_t, double)   \
+    F(I64TruncSatF64U, intTruncSat, double, uint64_t, uint64_t, double) \
+    F(F64PromoteF32, doConvert, float, double, double, float)           \
+    F(F32DemoteF64, doConvert, double, float, float, double)
+
+#define FOR_EACH_BYTECODE_LOAD_OP(F) \
+    F(I32Load, int32_t, int32_t)     \
+    F(I32Load8S, int8_t, int32_t)    \
+    F(I32Load8U, uint8_t, int32_t)   \
+    F(I32Load16S, int16_t, int32_t)  \
+    F(I32Load16U, uint16_t, int32_t) \
+    F(I64Load, int64_t, int64_t)     \
+    F(I64Load8S, int8_t, int64_t)    \
+    F(I64Load8U, uint8_t, int64_t)   \
+    F(I64Load16S, int16_t, int64_t)  \
+    F(I64Load16U, uint16_t, int64_t) \
+    F(I64Load32S, int32_t, int64_t)  \
+    F(I64Load32U, uint32_t, int64_t) \
+    F(F32Load, float, float)         \
+    F(F64Load, double, double)
+
+#define FOR_EACH_BYTECODE_STORE_OP(F) \
+    F(I32Store, int32_t, int32_t)     \
+    F(I32Store16, int32_t, int16_t)   \
+    F(I32Store8, int32_t, int8_t)     \
+    F(I64Store, int64_t, int64_t)     \
+    F(I64Store32, int64_t, int32_t)   \
+    F(I64Store16, int64_t, int16_t)   \
+    F(I64Store8, int64_t, int8_t)     \
+    F(F32Store, float, float)         \
+    F(F64Store, double, double)
+
 class ByteCode {
 public:
-#if defined(WALRUS_ENABLE_COMPUTED_GOTO)
-    OpcodeKind opcode() const
-    {
-        return static_cast<OpcodeKind>(g_opcodeTable.m_addressToOpcodeTable[m_opcodeInAddress]);
-    }
-#else
-    OpcodeKind opcode() const
-    {
-        return m_opcode;
-    }
-#endif
+    enum Opcode : size_t {
```

Comment:
In jit we changed this to uint32_t. Do we need this large type?

Reply:
Fixed the enum value to `uint32_t`. Thanks